### ISSUE TITLE
Make MCMC kernels usable in a parallel tempering setting

### ISSUE THREAD

**Issue by fehiepsi** [pyro-ppl/numpyro#978]
This can be a follow-up work after #898, which makes Gibbs kernels composable. Parallel tempering is popular in applications due to its ability to address multimodal issues (related #643) so it would be nice to have a design that users can use MCMC kernels seamlessly in parallel tempering.

### Reference
+ _Dynamic temperature selection for parallel tempering in Markov chain Monte Carlo simulations_, W. D. Vousden, W. M. Farr, I. Mandel [link](https://academic.oup.com/mnras/article/455/2/1919/1118710)

### Motivation
+ PT is used in [Science's 2019 breakthrough of the year: The first image of a black hole"](https://www.aaas.org/news/sciences-2019-breakthrough-first-image-supermassive-black-hole) :D

**Comment by fehiepsi** [comment]
It turns out that TFP has [ReplicaExchangeMC](https://www.tensorflow.org/probability/api_docs/python/tfp/mcmc/ReplicaExchangeMC) that users can use with NumPyro models through our [wrappers](http://num.pyro.ai/en/stable/mcmc.html#replicaexchangemc). So I would like to close this issue to avoid unnecessary efforts.
